src/BlockTemplatesController.php

```diff
@@ -429,8 +430,12 @@ function( $classes ) {
 							}
 						}
 
-						$new_content       = SingleProductTemplateCompatibility::add_compatibility_layer( $template->content );
-						$template->content = $new_content;
+						if ( post_password_required() ) {
+							$template->content = SingleProductTemplate::add_password_form( $template->content );
+						} else {
+							$new_content       = SingleProductTemplateCompatibility::add_compatibility_layer( $template->content );
+							$template->content = $new_content;
```

Comment:
Could we not have just done:

```php
$template->content = SingleProductTemplateCompatibility::add_compatibility_layer( $template->content );
```

Reply:
True! I didn't change that code, but I just moved! Should I create a follow-up PR?

Reply:
I don't think it's urgent. I think I'll create a follow up for it later and you can review if that works 😃